grakn-graql/src/main/java/ai/grakn/graql/internal/query/ConceptBuilder.java

```diff
@@ -314,21 +349,47 @@ private OntologyConcept putOntologyConcept() {
         OntologyConcept superConcept = expect(SUPER_CONCEPT);
         Label label = expect(LABEL);
 
+        OntologyConcept concept;
+
         if (superConcept.isEntityType()) {
-            return executor.graph().putEntityType(label).sup(superConcept.asEntityType());
+            concept = executor.graph().putEntityType(label);
         } else if (superConcept.isRelationType()) {
-            return executor.graph().putRelationType(label).sup(superConcept.asRelationType());
+            concept = executor.graph().putRelationType(label);
         } else if (superConcept.isRole()) {
-            return executor.graph().putRole(label).sup(superConcept.asRole());
+            concept = executor.graph().putRole(label);
         } else if (superConcept.isResourceType()) {
             ResourceType resourceType = superConcept.asResourceType();
             ResourceType.DataType<?> dataType = expectOrDefault(DATA_TYPE, resourceType.getDataType());
-            return executor.graph().putResourceType(label, dataType).sup(resourceType);
+            concept = executor.graph().putResourceType(label, dataType);
         } else if (superConcept.isRuleType()) {
-            return executor.graph().putRuleType(label).sup(superConcept.asRuleType());
+            concept = executor.graph().putRuleType(label);
         } else {
             throw GraqlQueryException.insertMetaType(label, superConcept);
         }
+
+        setSuper(concept, superConcept);
+
+        return concept;
     }
 
+    /**
+     * Make the second argument the super of the first argument
+     *
+     * @throws GraqlQueryException if the types are different, or setting the super to be a meta-type
+     */
+    public static void setSuper(OntologyConcept subConcept, OntologyConcept superConcept) {
```

Comment:
Oh bugger. This is done because the setters are only on the leaf interfaces . . . I am wondering if maybe graql can start getting access to some internal methods? At the end of the day this setter is actually implemented in `OntologyConceptImpl`